src/erlcloud_ec2.erl

```diff
@@ -1941,44 +2057,84 @@ describe_snapshot_attribute(SnapshotID, create_volume_permission, Config)
             Error
     end.
 
-%%
-%%
--spec describe_snapshots() -> [proplist()].
-describe_snapshots() -> describe_snapshots([], "self").
+-type ec2_snapshot_ids() :: [string()].
+-type ec2_snapshot_owner() :: string().
+-type ec2_snapshot_restorable_by() :: string() | none.
```

Comment:
Similar style-question. I'm used to putting all my type declarations at the top of the file, even when some are specific to a few private functions.